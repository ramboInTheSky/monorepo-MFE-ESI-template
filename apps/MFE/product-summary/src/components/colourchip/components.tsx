import styled from "styled-components"
import { Link } from "@mui/material"

export const ColourChipListItem = styled.li`
    box-sizing: border-box;
    margin-right: 0.125rem;
    margin-bottom: 0.125rem;
    border: 0.125rem solid #fff;
    border-radius: ${({ theme }) => theme.colours.utilities.chipBorderRadius};
    max-height: 2.25rem;
    max-width: 2.25rem;
    width: calc((100% - 1rem) / 9);
    border-width: 0.0625rem;
    position: relative;

    &.selected {
        border: ${props => props.theme.colours.utilities.activeBorder};

        &:before {
            border-radius: ${({ theme }) => `calc(${theme.colours.utilities.chipBorderRadius} - .125rem)`};
            border: 0.0625rem solid #fff;
            box-sizing: border-box;
            content: "";
            display: block;
            height: 100%;
            position: absolute;
            pointer-events: none;
            width: 100%;
            z-index: 1;
        }
    }
`

export const ColourChipLink = styled(Link)`
    padding-top: 100%;
    border-radius: ${props => props.theme.colours.utilities.chipBorderRadius};
    position: relative;
    display: block;
    // Color value to be part of theme object - PBI23328
    background: #d1d1d1;

    > img {
        max-height: 100%;
        max-width: 100%;
        border-radius: ${props => props.theme.colours.utilities.chipBorderRadius};
        position: absolute;
        top: 0;
        left: 0;
    }
`
