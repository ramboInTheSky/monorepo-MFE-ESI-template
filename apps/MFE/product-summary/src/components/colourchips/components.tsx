import { Collapse } from '@mui/material';
import {breakpoints} from "@monorepo/themes"
import styled from "styled-components"

export const ColourChipCollapse = styled(Collapse)`
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
`

export const ColourChipViewButtonContainer = styled.div``

export const ColourChipsList = styled.ul`
    list-style-type: none;
    display: flex;
    flex-wrap: wrap;
    padding: 0;
    overflow: hidden;
    max-width: 13.5rem;
    margin: 0;
    ${({theme}) =>
        theme.enablingCenteredContent &&
        `
        justify-content: center;
        margin-left: 0.75rem;
    `}
    > li:nth-child(9n) {
        margin-right: 0;
    }
`

export const ColourChipsContainer = styled.div`
    @media (max-width: ${breakpoints.values.lg - 1}px) {
        display: none;
    }

    > div:nth-child(2n) {
        display: ${({theme}) => `${theme.enablingCenteredContent ? "inline-flex" : "flex"}`};
        margin-top: 0 0.5rem 0 0.5rem;
        cursor: pointer;
    }

    > div:nth-child(2n) > button {
        border: 0;
        padding: 0;
        background: none;
        margin-right: 0.5rem;
        color: ${props => props.theme.colours.text.hyperlink};
        font-family: ${({theme}) => `${theme.colours.font.primary.medium.family}, ${theme.colours.font.default}`};
        margin-right: 0.5rem;
        font-size: 0.875rem;
        cursor: pointer;
        line-height: 1.5;

        > img {
            margin-left: 0.5rem;
        }
    }
`
