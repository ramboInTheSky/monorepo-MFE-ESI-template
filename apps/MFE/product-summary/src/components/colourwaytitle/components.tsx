import styled from "styled-components"
import {Link} from "@mui/material"
import {breakpoints} from "@monorepo/themes"

export const TileTitle = styled(Link)`
    height: 1.5em;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    display: block;
    text-transform: capitalize;
    text-decoration: none;

    @media (max-width: ${breakpoints.values.lg - 1}px) {
        height: 3em;
        white-space: normal;
        -webkit-line-clamp: 2;
        -webkit-box-orient: vertical;
        display: -webkit-box;
        margin-bottom: 0.25rem;

        &:after {
            position: relative;
            content: "";
            border-bottom: 0.0625rem solid #d1d1d1;
            width: 1rem;
            margin: auto;
            display: block;
            bottom: -0.8rem;
        }
    }
`
