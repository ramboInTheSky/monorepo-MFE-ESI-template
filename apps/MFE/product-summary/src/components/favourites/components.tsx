import styled from "styled-components"
import {FavouriteState} from "../../models/Favourites"
import env from "../../config/env"

interface FavouriteProps {
    favState: string
    shouldAnimate: boolean
}

export const FavouritesContainer = styled.div`
    position: absolute;
    bottom: 1.125rem;
    right: 0.625rem;
    border: ${props => props.theme.colours.utilities.dividerDark};
    background: #fff;
    border-radius: 50%;
    cursor: pointer;
    z-index: 100;
    height: 2.25rem;
    width: 2.25rem;
`

export const FavouritesIcon = styled.svg<FavouriteProps>`
    && {
        /* transform:rotate added to prevent pixel snapping and rectify visible jump in animation*/
        /* https://stackoverflow.com/questions/22097660/is-there-a-way-i-can-force-chrome-to-do-subpixel-rendering-for-a-slow-translatio
https://johnresig.com/blog/sub-pixel-problems-in-css/ */
        transform: rotateZ(-0.0000000001deg);
        width: 100%;
        height: 100%;

        ${AnimationProps =>
            AnimationProps.favState === FavouriteState.Inactive &&
            `

                @media all and (-ms-high-contrast: none) {
                    background-image: url(${env.REACT_APP_BLOB_STORAGE_PATH}/icons/shared/favourites-inactive.png); 
                    background-repeat: no-repeat;
                    background-size: 100% 100%;
                }
        `}

        ${FavouriteProps =>
            FavouriteProps.favState === FavouriteState.Loading &&
            `
            stroke-dasharray: 40;
            stroke-dashoffset: 80;
            stroke-linecap: round;
            animation: favourite-loading 0.8s linear infinite reverse;

            @keyframes favourite-loading {
                0% {
                    stroke-dashoffset: 80;
                }
                100% {
                    stroke-dashoffset: 160;
                }
            }

            @media all and (-ms-high-contrast: none) {
                background-image: url(${env.REACT_APP_BLOB_STORAGE_PATH}/icons/shared/favourites-loading.gif); 
                background-repeat: no-repeat;
                background-size: 100% 100%;
            }
        `}

        ${FavouriteProps =>
            FavouriteProps.favState === FavouriteState.Active &&
            FavouriteProps.shouldAnimate &&
            `
            animation: favourite-loaded 1s ease-in-out;
            
            @keyframes favourite-loaded {
                0% {
                    transform: scale(1);
                }
                50% {
                    transform: scale(1.25);
                }
                100% {
                    transform: scale(1);
                }
            }
            
        `}

        ${FavouriteProps =>
            FavouriteProps.favState === FavouriteState.Active &&
            `
            @media all and (-ms-high-contrast: none) {
                background-image: url(${env.REACT_APP_BLOB_STORAGE_PATH}/icons/shared/favourites-active.png); 
                background-repeat: no-repeat;
                background-size: 100% 100%;
            }
            
        `}
    }
`

export const FavouritesButton = styled.button`
    && {
        padding: 0.5rem 0.5rem;
        height: 100%;
        width: 100%;
        background: none;
        border: none;
        cursor: pointer;
    }
`
