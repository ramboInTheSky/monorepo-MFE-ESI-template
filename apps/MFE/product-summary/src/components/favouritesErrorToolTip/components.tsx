import styled from "styled-components"

const buttonOverrides = `
    -webkit-appearance: none;
    appearance: none;
    background: none;
    border: 0;
    cursor: pointer;
    padding: 0;
`

export const Container = styled.div`
    padding: 1rem;
    width: 18rem;
    border: ${props => props.theme.colours.popover.border};
    background: ${props => props.theme.colours.header.myAccount.background};
    border-radius: ${props => props.theme.colours.popover.radius};
    box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.4);
`

export const ErrorContainer = styled(Container)`
    width: 8.9rem;
    text-align: center;
`

export const Header = styled.div`
    margin-bottom: 1rem;
    font-family: ${({theme}) => `${theme.colours.font.primary.regular.family}, ${theme.colours.font.default}`};
    display: flex;
    justify-content: space-between;
`

export const Title = styled.div`
    font-family: ${({theme}) => `${theme.colours.font.primary.medium.family}, ${theme.colours.font.default}`};
    font-size: 0.938rem;
    font-weight: 500;
`

export const CloseButton = styled.button`
    ${buttonOverrides}
    height: 0.75rem;
    width: 0.75rem;
`

export const CloseImage = styled.img`
    width: 0.75rem;
    height: 0.75rem;
`

export const Content = styled.div`
    font-size: 0.875rem;
    line-height: 1.313rem;
`

export const ShoppingLinkWrapper = styled.div`
    text-align: center;
`

export const ShoppingLink = styled.button`
    ${buttonOverrides}
    color: ${props => props.theme.colours.text.hyperlink};
    font-family: ${({theme}) => `${theme.colours.font.primary.regular.family}, ${theme.colours.font.default}`};
    font-size: 0.875rem;
    text-decoration: underline;
`

export const ManageFavouritesButton = styled.div`
    height: 2.25rem;
    margin: 1.1rem 0 1.1rem;
    & > a {
        text-decoration: none;
        width: 100%;
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 0 !important;
        font-size: 1rem !important;
        font-weight: 500 !important;
        visibility: visible !important;
    }
`