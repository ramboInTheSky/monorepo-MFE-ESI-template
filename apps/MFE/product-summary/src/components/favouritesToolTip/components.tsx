import styled from "styled-components"

export const Container = styled.div`
    padding: 1rem;
    min-width: 18rem;
    border: ${props => props.theme.colours.popover.border};
    background: ${props => props.theme.colours.header.myAccount.background};
    border-radius: ${props => props.theme.colours.popover.radius};
    box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.4);
`

export const Header = styled.div`
    margin-bottom: 1rem;
    font-family: ${({theme}) => `${theme.colours.font.primary.regular.family}, ${theme.colours.font.default}`};
    display: flex;
    justify-content: space-between;
`

export const Title = styled.div`
    font-size: 0.938rem;
`

export const CloseButton = styled.button`
    -webkit-appearance: none;
    appearance: none;
    background: none;
    border: 0;
    cursor: pointer;
    padding: 0;
    height: 0.75rem;
    width: 0.75rem;
`

export const CloseImage = styled.img`
    width: 0.75rem;
    height: 0.75rem;
`

export const Content = styled.div`
    font-size: 0.875rem;
`

export const Link = styled.a`
    color: ${props => props.theme.colours.text.hyperlink};
`

export const UserSection = styled.p`
    margin-bottom: 0.2rem;
`
