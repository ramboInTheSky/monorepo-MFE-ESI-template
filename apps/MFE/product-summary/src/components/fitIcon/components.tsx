import styled from "styled-components"

export const StyledImage = styled.img`
    margin-bottom: 0.5rem;
    width: 1rem;
    height: 1rem;
`

export const FitsContainer = styled.div`
    position: absolute;
    left: 0.5rem;
    bottom: 0rem;
    max-width: 1rem;
    z-index: 2;
`
