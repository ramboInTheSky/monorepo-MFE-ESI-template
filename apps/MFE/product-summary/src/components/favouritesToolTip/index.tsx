import React, {useEffect, useState} from "react"
import {useModalsCloseObservable} from "@monorepo/eventservice"
import ToolTip from "@monorepo/tooltipv2"
import {Container, Title, CloseButton, CloseImage, Header, Content, Link, UserSection} from "./components"
import env from "../../config/env"
import {
    getHardLogoutPromptDisplayed,
    getIdentifiedUser,
    getFavouriteLoginPromptDisplayed,
    getSoftLoginFirstname,
} from "../../utils/getWindowNextFavourites"
import {FavouriteState} from "../../models/Favourites"
import {IS_BROWSER} from "../../utils/window"
import {getAccountPage, getRegisterPage} from "../../urls"
import {TextModel} from "../../models/Text"

export interface FavoritesToolTipProps {
    baseUrl: string
    isProductFavourited: boolean
    favState: FavouriteState
    referenceElement: HTMLElement | null
    text: TextModel
    setFavClicked: (value: boolean) => void
}

const FavoritesToolTip = ({
    baseUrl,
    isProductFavourited,
    favState,
    referenceElement,
    text,
    setFavClicked,
}: FavoritesToolTipProps) => {
    const [toolTipVisible, setToolTipVisible] = useState(false)

    const showAuthenticatedToolTip = IS_BROWSER() && !getHardLogoutPromptDisplayed() && getIdentifiedUser()
    const showAnonymousToolTip = IS_BROWSER() && !getFavouriteLoginPromptDisplayed() && !getSoftLoginFirstname()

    const getContent = () => {
        if (showAuthenticatedToolTip) {
            return (
                <Content data-testid="product-summary-favourite-authenticated-tooltip">
                    {text.labels.favouritesAddedInfo}
                    <br /> {getSoftLoginFirstname()}
                    {text.labels.apostrophe} <strong>{text.labels.favouritesList}</strong>
                    <UserSection>
                        {text.labels.not} {getSoftLoginFirstname()}
                        {text.labels.questionSymbol} <Link href={getAccountPage(baseUrl)}>{text.buttons.signOut}</Link>
                    </UserSection>
                </Content>
            )
        }

        if (showAnonymousToolTip) {
            return (
                <Content data-testid="product-summary-favourite-anonymous-tooltip">
                    <Link href={getAccountPage(baseUrl)}>{text.buttons.signIn}</Link> {text.labels.or}{" "}
                    <Link href={getRegisterPage(baseUrl)}>{text.buttons.register}</Link>{" "}
                    {text.labels.favouritesSavePermanently}
                </Content>
            )
        }
    }

    useEffect(() => {
        if (isProductFavourited && favState === FavouriteState.Active) {
            if (showAuthenticatedToolTip || showAnonymousToolTip) {
                setToolTipVisible(true)
            }
        }
    }, [isProductFavourited, favState, showAuthenticatedToolTip, showAnonymousToolTip])

    const handleClose = () => {
        setToolTipVisible(false)
        setFavClicked(false)
    }

    useModalsCloseObservable(handleClose)

    if (!toolTipVisible) return null
    return (
        <ToolTip handleClose={handleClose} referenceElement={referenceElement}>
            <Container>
                <Header>
                    <Title>{text.product.favourites}</Title>
                    <CloseButton onClick={handleClose} data-testid="product-summary-favourite-tooltip-close">
                        <CloseImage
                            src={`${env.REACT_APP_BLOB_STORAGE_PATH}/icons/shared/close.svg`}
                            alt={text.labels.continueShopping}
                        />
                    </CloseButton>
                </Header>
                {getContent()}
            </Container>
        </ToolTip>
    )
}

export default FavoritesToolTip
