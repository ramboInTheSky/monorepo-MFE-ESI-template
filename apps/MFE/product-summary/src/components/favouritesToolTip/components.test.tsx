import React from "react"
import {render} from "@testing-library/react"
import {ThemeProvider} from "styled-components"
import {mockTheme} from "../../../__mocks__/mockStore"
import {Container, Header, Title, CloseButton, Content, Link, UserSection} from "./components"

describe("Given FavouritesToolTip", () => {
    describe("Given Container", () => {
        it("should render as expected", () => {
            const {asFragment} = render(
                <ThemeProvider theme={mockTheme}>
                    <Container />
                </ThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
    })

    describe("Given Header", () => {
        it("should render as expected", () => {
            const {asFragment} = render(
                <ThemeProvider theme={mockTheme}>
                    <Header />
                </ThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
    })

    describe("Given Title", () => {
        it("should render as expected", () => {
            const {asFragment} = render(
                <ThemeProvider theme={mockTheme}>
                    <Title />
                </ThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
    })

    describe("Given CloseButton", () => {
        it("should render as expected", () => {
            const {asFragment} = render(
                <ThemeProvider theme={mockTheme}>
                    <CloseButton />
                </ThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
    })

    describe("Given Content", () => {
        it("should render as expected", () => {
            const {asFragment} = render(
                <ThemeProvider theme={mockTheme}>
                    <Content />
                </ThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
    })

    describe("Given Link", () => {
        it("should render as expected", () => {
            const {asFragment} = render(
                <ThemeProvider theme={mockTheme}>
                    <Link href="https://www.amido.com" />
                </ThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
    })

    describe("Given UserSection", () => {
        it("should render as expected", () => {
            const {asFragment} = render(
                <ThemeProvider theme={mockTheme}>
                    <UserSection />
                </ThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
    })
})
