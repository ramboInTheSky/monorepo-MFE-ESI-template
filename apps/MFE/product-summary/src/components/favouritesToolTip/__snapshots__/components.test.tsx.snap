// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given FavouritesToolTip Given CloseButton should render as expected 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border: 0;
  cursor: pointer;
  padding: 0;
  height: 0.75rem;
  width: 0.75rem;
}

<button
    class="c0"
  />
</DocumentFragment>
`;

exports[`Given FavouritesToolTip Given Container should render as expected 1`] = `
<DocumentFragment>
  .c0 {
  padding: 1rem;
  min-width: 18rem;
  border: 0.0625rem solid #9e9e9e;
  background: #fff;
  border-radius: 0.25rem;
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.4);
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`Given FavouritesToolTip Given Content should render as expected 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 0.875rem;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`Given FavouritesToolTip Given Header should render as expected 1`] = `
<DocumentFragment>
  .c0 {
  margin-bottom: 1rem;
  font-family: "AzoSansRegular","Helvetica Neue",Arial,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`Given FavouritesToolTip Given Link should render as expected 1`] = `
<DocumentFragment>
  .c0 {
  color: #257f39;
}

<a
    class="c0"
    href="https://www.amido.com"
  />
</DocumentFragment>
`;

exports[`Given FavouritesToolTip Given Title should render as expected 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 0.938rem;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`Given FavouritesToolTip Given UserSection should render as expected 1`] = `
<DocumentFragment>
  .c0 {
  margin-bottom: 0.2rem;
}

<p
    class="c0"
  />
</DocumentFragment>
`;
